cmd/snap-repair/runner.go

```diff
@@ -900,8 +906,48 @@ func (run *Runner) Applicable(headers map[string]interface{}) bool {
 		}
 	}
 
-	// TODO:UC20: need to consider filtering by bases and modes in the assertion
-	// here
+	// also filter by base snaps and modes
+	bases, err := stringList(headers, "bases")
+	if err != nil {
+		return false
+	}
+
+	if len(bases) != 0 && !strutil.ListContains(bases, run.state.Device.Base) {
+		return false
+	}
+
+	modes, err := stringList(headers, "modes")
+	if err != nil {
+		return false
+	}
+
+	// modes is slightly more nuanced, if the modes setting in the assertion
+	// header is unset, that means that the assertion should only be run in
+	// "run" mode on uc20
```

Comment:
this comment sounds wrong or I'm misunderstanding it

Reply:
yeah this is confusing, I will rephrase